src/Security/Provider/UserProvider.php

```diff
@@ -12,118 +12,59 @@
-        $previousUser = $this->userManager->findUserBy(['githubId' => $username]);
-
-        /** @var User $user */
-        $user->setGithubId($username);
-        $user->setGithubToken($response->getAccessToken());
-        $user->setGithubScope($response->getOAuthToken()->getRawToken()['scope']);
-
-        // The account is already connected. Do nothing
-        if ($previousUser === $user) {
-            return;
-        }
-
-        // 'disconnect' a previous account
-        if (null !== $previousUser) {
-            $previousUser->setGithubId(null);
-            $previousUser->setGithubToken(null);
-            $this->userManager->updateUser($previousUser);
-        }
-
-        $this->userManager->updateUser($user);
-
-        $this->scheduler->scheduleUserScopeMigration($user->getId(), '', $user->getGithubScope());
-    }
-
-    public function loadUserByOAuthUserResponse(UserResponseInterface $response)
-    {
-        $username = $response->getUsername();
-        if (!$username || $username <= 0) {
-            throw new \LogicException('Failed to load info from GitHub');
-        }
-
-        /** @var User $user */
-        $user = $this->userManager->findUserBy(['githubId' => $username]);
-
-        if (!$user) {
-            throw new AccountNotLinkedException(sprintf('No user with github username "%s" was found.', $username));
-        }
-
-        if ($user->getGithubId() !== (string) $response->getUsername()) {
-            throw new \LogicException('This really should not happen but checking just in case');
-        }
-
-        if ($user->getGithubToken() !== $response->getAccessToken()) {
-            $user->setGithubToken($response->getAccessToken());
-            $oldScope = $user->getGithubScope();
-            $user->setGithubScope($response->getOAuthToken()->getRawToken()['scope']);
-            $this->userManager->updateUser($user);
-            if ($oldScope !== $user->getGithubScope()) {
-                $this->scheduler->scheduleUserScopeMigration($user->getId(), $oldScope ?: '', $user->getGithubScope());
-            }
-        }
-
-        return $user;
-    }
-
     /**
      * {@inheritDoc}
      */
     public function loadUserByUsername($usernameOrEmail)
     {
-        return $this->userProvider->loadUserByUsername($usernameOrEmail);
+        return $this->getRepo()->findOneByUsernameOrEmail((string) $usernameOrEmail);
     }
 
     /**
      * {@inheritDoc}
      */
     public function refreshUser(UserInterface $user)
     {
-        return $this->userProvider->refreshUser($user);
+        if (!$user instanceof User) {
+            throw new \UnexpectedValueException('Expected '.User::class.', got '.get_class($user));
+        }
+
+        return $this->getRepo()->find($user->getId());
     }
 
     /**
      * {@inheritDoc}
      */
     public function supportsClass($class)
     {
-        return $this->userProvider->supportsClass($class);
+        return $class === User::class;
```

Comment:
this could break if the class of the serialized token is the doctrine proxy class rather than the entity class directly.